problems/numerical_diff.py

```diff
@@ -0,0 +1,77 @@
+import logging
+from math import exp, sin
+from random import uniform
+
+from problems.test_case import TestCase, TestCaseTypeEnum
+from problems.solutions.numerical_diff import numerical_diff
+
+logger = logging.getLogger(__name__)
+
+FUNCTION_NAME = "numerical_diff"
+INPUT_VARS = ['f', 'x']
+OUTPUT_VARS = ['df']
+
+STATIC_RESOURCES = []
+
+PHYSICAL_CONSTANTS = {}
+ATOL = {}
+RTOL = {
+    'df': 0.005,  # 0.5% relative tolerance
+}
+
+
+class TestCaseType(TestCaseTypeEnum):
+    ZERO = ("Zero function", 1)
+    CONSTANT = ("Constant function", 1)
+    LINEAR = ("Linear polynomial", 1)
+    QUADRATIC = ("Quadratic polynomial", 1)
+    CUBIC = ("Cubic polynomial", 1)
+    SINE = ("Sine function", 1)
+    EXPONENTIAL = ("Exponential function", 1)
+
+
+class ProblemTestCase(TestCase):
+    def input_tuple(self) -> tuple:
+        return self.input['f', 'x'],
```

Comment:
Ah not sure if this is valid syntax. Did you mean something like this?
```suggestion
        return (self.input['f'], self.input['x'])
```

Reply:
Yeah, I believe that is not a valid syntax. But I was just copying from the other problems' script – they used the same syntax – so I thought it was something a must to do. See for example [here](https://github.com/project-lovelace/lovelace-problems/blob/main/problems/almost_pi.py).

Reply:
Ah sorry I should have probably made the code clearer. I think for `almost_pi.py` since there is only one input and `input_tuple` returns a `tuple` then we ended up writing

```python
def input_tuple(self) -> tuple:
    return self.input['N'],  # Return a tuple of length 1, same as return (self.input['N'],)
```

and it seems that here you want to return 2 inputs as a tuple of length 2 so I suggested

```python
def input_tuple(self) -> tuple:
    return (self.input['f'], self.input['x'])
```

I guess this is what you meant with `return self.input['f', 'x']`? I haven't seen this syntax for indexing multiple elements of a dictionary but would be cool if it works!

Reply:
Ah yes, I do mean `(self.input['f'], self.input['x'])`! My bad, I'm still getting used to the codebase, I forgot that that line is accessing a dictionary via a key.

Reply:
Ah haha no worries. I made a GitHub suggestion that you can commit by clicking on "Commit suggestion" if you agree with it.

![image](https://user-images.githubusercontent.com/20099589/115413955-9d9def00-a1c3-11eb-9eea-bcc0e0042990.png)

You can also keep committing to this branch (`numerical-diff`) and it should show up in this PR.

Reply:
Okay, I've executed "Commit suggestion" and it works like a charm now! Thanks for your help.